### ISSUE TITLE
[doc] Figure&nbsp;2:  One cluster for central services, one for SAP HANA SR 

### ISSUE THREAD

**Issue by dkloos** [SUSE/suse-best-practices#132]
Hello,
while reading trought this Guide I was "struggeling" with this design.

Figure&nbsp;2:  One cluster for central services, one for SAP HANA SR :
[SAP S/4 HANA - Enqueue Replication 2](https://documentation.suse.com/sbp/all/html/SAP_S4HA10_SetupGuide-SLE15/index.html#id-1.4.10.3.3)
![image](https://user-images.githubusercontent.com/44433238/99955759-28013480-2d85-11eb-9fe5-60ec27626f12.png)

The PAS and AAS instances are installed on the DB Cluster Node's. 
In Chapter 3.2 I read following statment:

> ...The database and SAP dialog instances can be added to the cluster either by adding more nodes to the cluster or by installing the database on either of the existing nodes. **However, it is recommended to install the database on a separate cluster**.

So that is my understanding:
![image](https://user-images.githubusercontent.com/44433238/99955444-a6110b80-2d84-11eb-87d9-2aebc0734b61.png)

Or at least give some hint's what are the benefit's of installing the PAS and AAS on the DB Cluster rather then on the ASCS Cluster.

**Comment by chabowski** [comment]
Hi @dkloos - thank you very much for your comment! 

@fmherschel Would you mind having a look at the issue regarding the paper "SUSE Linux Enterprise Server for SAP Applications 15 - SAP S/4 HANA - Enqueue Replication 2 High Availability Cluster - Setup Guide"

Thanks again!
Meike

**Comment by fmherschel** [comment]
It is and was a SAP certification requirement. And the guide referenced by the certification needs to describe exactly what the architecture for the certified cluster is/was:
Why does SAP recommend to have PAS and AAS separately? It is that if you install PAS and AAS on the same
cluster nodes that implies test interference if you kill a cluster node an Application server PLUS an SAP Instance for CS or Replication are dying in parallel. So the described test results and procedures are only fitting, if those components are separated.
As the certification does not test the outage of SAP HANA cluster nodes you can install the PAS/AAS on the same nodes to geta minimum lab environment.
Does that help?

**Comment by dkloos** [comment]
Hello @fmherschel ,
Thanks for your answer.
So the reason why the PAS was not installed on the same cluster as the  ASCS Service is just to keep the focus on the cluster setup and certification? So there is no concern installing the PAS (or AAS) on the same nodes as the ASCS rather then on a HDB Server?

**Comment by fmherschel** [comment]
@dkloos 
Yes and no ;-) Initially the certification setup did "force" us to split between the ASCS/ERS vs PAS and AAS. You could combine then I would say. But combining might also increase complexity. I would prefer to have to "tiny" VM systems in a cluster which are running ASCS and ERS and to have the workload (Application Servers) separated.
And ... if you would follow our described tests you need to adapt the expectations (results), because when you kill a node in a cluster running ASCS/ERS PLUS PAS and AAS you would kill always TWO SAP components.
You would not pass the SAP certification tests.

**Comment by pirat013** [comment]
@dkloos 
one remark in addition to @fmherschel last statement. If you install the PAS and AAS on the same host's as ASCS and ERS are installed and they run out side of the cluster. If one of you cluster node's failed and need to be fenced you will lose your PAS / AAS together with your ASCS / ERS. The ASCS / ERS is maintained by pacemaker and will be moved to a node who is alive.
PAS and AAS in a cluster setup is possible but there is one point to consider here. The soft shutdown of an SAP instance is not possible with pacemaker. That means a PAS or AAS controlled by pacemaker will close open transaction immediately in case of a PAS / AAS shutdown.

**Comment by dkloos** [comment]
@fmherschel 
> But combining might also increase complexity. I would prefer to have to "tiny" VM systems in a cluster which are running ASCS and ERS and to have the workload (Application Servers) separated.

That is true and I fully agree with you!!  
So to put the PAS and AAS Service (which are not managed by pacemaker ressources) on the HDB Cluster is not a "....perfect Enhancment..." if set up like this picture??? [SAP S/4 HANA - Enqueue Replication 2](https://documentation.suse.com/sbp/all/html/SAP_S4HA10_SetupGuide-SLE15/index.html#id-1.4.10.3.3)


@pirat013 
> If you install the PAS and AAS on the same host's as ASCS and ERS are installed and they run out side of the cluster.

Thats clear for me and a good hint. This hint should be included under that proposed solution / image ?! [SAP S/4 HANA - Enqueue Replication 2](https://documentation.suse.com/sbp/all/html/SAP_S4HA10_SetupGuide-SLE15/index.html#id-1.4.10.3.3)

This "architecture image" is combining the "HDB Cluster" with the PAS / AAS Instance (not clustered). So I gues this will behave the same way as putting it on the ASCS/ERS Cluster (no Soft Shutdown + kill of PAS or AAS if stonith is triggerd by the cluster).

So my initial question: 
"_Or at least give some hint's what are the benefit's of installing the PAS and AAS on the DB Cluster rather then on the ASCS Cluster."_.
..is still unclear for me.

THANK you both for your advice :)

**Comment by fmherschel** [comment]
I like to stop this discussion here at github as it is to public for me.

**Comment by fmherschel** [comment]
> "Or at least give some hint's what are the benefit's of installing the PAS and AAS on the DB Cluster rather then on the ASCS Cluster.".

It is still a point of standardization. The SAP certification extracts them with good reasons as already explained. What your point might be - we should not have even added them to the second cluster. Does that answer your question?

**Comment by dkloos** [comment]
@fmherschel : Yes, That is the point I want to clarify. Cause for me it was kind of confusing while searching for a best practice design for an HA HANA DB and also ASCS setup. 
The reason why this setup was done for the "Lab Setup" is now clear for me. Thanks!
And my conclusion of that discussion is:
- Dialog instances should not be on clustered Systems -> they will be killed hard on cluster failure
- Cluster test will not work as expected if Dialog instance is on the same host -> Double failure. 
- The PAS + AAS instance are not part of the cluster ( which is not visible in that image )

Thank you for the assistance to clarify that. I will need to discuss the pro / cons with my colleagues. 

Thank you both for your engagment!

**Comment by fmherschel** [comment]
Thanks for the feedback, @dkloos . I interpret this issue (question) as answered, so I close the issue right now.
Please feel free to open an other issue.

**Comment by fmherschel** [comment]
Thanks for the feedback, @dkloos . I interpret this issue (question) as answered, so I close the issue right now.
Please feel free to open an other issue.
